### ISSUE TITLE
group_interval affects context timeout for notifications

### ISSUE THREAD

**Issue by dgl** [prometheus/alertmanager#2486]
**What did you do?**

Configured a short `group_interval` to get faster reaction to alerts:

```
group_interval: 5s
```

**What did you expect to see?**

Notifications delivered more quickly than the default group interval. (Our webhook receiver deals with grouping itself, so these don’t become spammy notifications a user has to deal with, they are just two systems updating state between each other).

**What did you see instead? Under which circumstances?**

Sometimes the notifications weren’t delivered, or the HTTP request was aborted part way through processing, if the webhook receiver was slow.

This appears to be because of:
https://github.com/prometheus/alertmanager/blob/master/dispatch/dispatch.go#L392-L394

```go
context.WithTimeout(ag.ctx, ag.timeout(ag.opts.GroupInterval))
```

It is somewhat surprising that group_interval is applied to the timeout, as nothing in the configuration documentation mentions it, or warns that setting group_interval to a short duration could cause unexpected behaviour.

(The timeout is bounded by MinTimeout, so 10s is the lowest it will go.)

Potentially this is fixable in the short term by updating the documentation and mentioning this as a configuration problem. However it would also be nice to be able to set group_interval to a short duration, but still get notifications delivered with a longer timeout.

**Comment by beorn7** [comment]
Yeah, that's weird. @simonpasquier do you have any context on this?

**Comment by grobinson-grafana** [comment]
Yes that's correct, Alertmanager will cancel any in progress notifications from the last `group_interval`. If it doesn't, then you may receive both duplicate and out of order notifications. For example, you could get your resolved notification before your firing notification.

**Comment by TheMeier** [comment]
IMHO that design is ok and we should add some documentation for it.
